pkg/backends/tcp.go

```diff
@@ -113,6 +115,7 @@ func newTCPSession(conn net.Conn, closeChan chan struct{}) *TCPSession {
 		conn:      conn,
 		framer:    framer.New(),
 		closeChan: closeChan,
+		once:      sync.Once{},
```

Comment:
Let's give it a more meaningful name like `closeChanCloser` or something ... calling it `once` gives no hint at all to future programmer/archaeologists digging through this code what its purpose might be.